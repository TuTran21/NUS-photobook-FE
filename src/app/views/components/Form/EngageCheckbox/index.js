/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-props-no-spreading */
import React, { useState, useCallback, useEffect } from 'react';
import styled from 'styled-components';
import theme from 'app/themes/styles';
import UnChecked from './UnChecked.svg';
import Checked from './Checked.svg';

const StyledCheckbox = styled.div`
	display: flex;
	align-items: center;
	justify-content: flex-start;
	cursor: pointer;
	margin-bottom: 0px;

	label {
		margin-bottom: 0 !important;
		margin-left: 5px;
		color: ${theme.ncpDark};
		cursor: pointer;
	}

	img {
		margin-top: -1px;
		cursor: pointer;
	}

	${props =>
		props.disabled
			? `
	  background-color: ${theme.ncpDisabledBackground};
	  padding: 5px 10px;
      border: none !important;
      opacity: 0.9;
  `
			: ''}
`;

const Label = styled.label`
    display: inline-block;
    margin-bottom: 0 !important;
    margin-left: 5px;
    cursor: pointer;
    line-height: 1.5;
    font-size: 16px;
}`;

export default function({ title, isChecked, onChange, disabled }) {
	const [checked, setChecked] = useState(isChecked);
	useEffect(() => {
		setChecked(isChecked);
	}, [isChecked]);

	const onCheck = useCallback(() => {
		if (disabled) {
			return;
		}
		setChecked(!checked);
		onChange(!checked);
	}, [checked, setChecked, onChange, disabled]);

	return (
		<StyledCheckbox onClick={onCheck} disabled={disabled}>
			{checked && <img src={Checked} alt="" />}
			{!checked && <img src={UnChecked} alt="" />}
			<Label>{` ${title}`}</Label>
		</StyledCheckbox>
	);
}
