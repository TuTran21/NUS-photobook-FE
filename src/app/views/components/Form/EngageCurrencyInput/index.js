/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-props-no-spreading */
import React from 'react';
import classnames from 'classnames';
import NumberFormat from 'react-number-format';

const Label = styled.label`
    display: inline-block;
    margin-bottom: 0 !important;
    cursor: pointer;
    line-height: 1.5;
    font-family: 'Nunito',Helvetica,Arial,sans-serif;
    font-size: 16px;
}`;

const FormGroup = styled.div``;

export default React.memo(({ title, value, name, onChange, required, disabled, currencyLabel }) => (
	<FormGroup className="eng-input--primary">
		<Label
			className={classnames('eng-label--input-title', {
				'eng-label--required': required,
			})}
		>
			{title}
		</Label>

		<NumberFormat
			value={value}
			name={name}
			disabled={disabled}
			thousandSeparator=","
			placeholder={currencyLabel ? `${currencyLabel} here` : '$ here'}
			className="form-control"
			prefix={currencyLabel || '$'}
			onValueChange={event => {
				onChange(event.floatValue);
			}}
		/>
	</FormGroup>
));
