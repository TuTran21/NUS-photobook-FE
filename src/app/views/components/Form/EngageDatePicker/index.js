/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable react/prop-types */
import React from 'react';
import DatePicker from 'react-datepicker';
import classnames from 'classnames';
import 'react-datepicker/dist/react-datepicker.css';
import moment from 'moment';
import CalendarIcon from './CalendarIcon.svg';

const Label = styled.label`
    display: inline-block;
    margin-bottom: 0 !important;
    cursor: pointer;
    line-height: 1.5;
    font-family: 'Nunito',Helvetica,Arial,sans-serif;
    font-size: 16px;
}`;

const FormGroup = styled.div``;

function EngageDatePicker({
	title,
	errors,
	onChange,
	value,
	name,
	required,
	disabled,
	minDate,
	placeholder,
	isClearable,
	showMonthYearPicker,
	...restProps
}) {
	return (
		<FormGroup className="eng-input--primary">
			<Label
				className={classnames('eng-label--input-title', {
					'eng-label--required': required,
				})}
			>
				{title}
			</Label>
			<DatePicker
				name={name}
				id={name}
				dateFormat="dd-MM-yyyy"
				className="form-control"
				selected={value}
				placeholderText={placeholder || 'Choose an option'}
				onChange={onChange}
				disabled={disabled}
				minDate={minDate ? moment(minDate).toDate() : moment().toDate()}
				isClearable={isClearable && !disabled}
				showMonthYearPicker={showMonthYearPicker}
				{...restProps}
			/>

			<img src={CalendarIcon} alt="" />
		</FormGroup>
	);
}

EngageDatePicker.defaultProps = {
	showMonthYearPicker: false,
};

export default React.memo(EngageDatePicker);
