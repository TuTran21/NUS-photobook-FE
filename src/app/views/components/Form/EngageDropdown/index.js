import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';
import styled from 'styled-components';
import Select, { components } from 'react-select';
// import SearchIcon from 'component\s/EngageIcons/SearchIcon';
import ArrowIcon from './Arrow.svg';

const Label = styled.label`
    display: inline-block;
    margin-bottom: 0 !important;
    cursor: pointer;
    line-height: 1.5;
    font-family: 'Nunito',Helvetica,Arial,sans-serif;
    font-size: 16px;
}`;

const FormGroupStyled = styled.div`
	margin-bottom: 0px;
	min-width: 130px;
	display: inline-block;
`;

/**
 *
 * @param {*} props { data,  }
 */
const DropdownIndicator = props => (
	<components.DropdownIndicator {...props}>
		<img alt="" src={ArrowIcon} />
	</components.DropdownIndicator>
);

const StyledSelect = styled(Select)``;

const EngageDropdown = React.memo(
	({
		title,
		errors,
		required,
		disabled,
		isClearable = false,
		placeholder,
		holderStyle,
		value,
		key,
		...restProps
	}) => (
		<FormGroupStyled style={holderStyle} className={errors ? 'errors' : ''}>
			{!!title && (
				<Label
					htmlFor={title}
					className={classnames('eng-label--input-title', {
						'eng-label--required': required,
					})}
				>
					{title}
				</Label>
			)}
			<StyledSelect
				key={key}
				value={value}
				className="eng-select--primary"
				classNamePrefix="eng-select--primary"
				styles={{
					placeholder: () => ({
						color: 'rgba(48, 51, 66, 0.8)',
					}),
				}}
				// menuIsOpen
				components={{ DropdownIndicator }}
				isClearable={isClearable && !disabled}
				placeholder={placeholder || 'Choose an option'}
				{...restProps}
				isDisabled={disabled}
			/>
		</FormGroupStyled>
	),
);

EngageDropdown.propTypes = {
	errors: PropTypes.any,
	title: PropTypes.any,
	required: PropTypes.any,
	labelClassName: PropTypes.any,
	selectClassName: PropTypes.any,
	isClearable: PropTypes.any,
	disabled: PropTypes.any,
	placeholder: PropTypes.any,
};
export default EngageDropdown;
