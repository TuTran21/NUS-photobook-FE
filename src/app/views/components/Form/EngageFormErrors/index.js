/* eslint-disable react/jsx-props-no-spreading */
import _ from 'lodash';
import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';
import theme from 'utils/theme';

import WarningIcon from './WarningIcon.svg';

const Container = styled.div`
	background-color: #fce8e6;
	border: 1px solid ${theme.engError};
	box-sizing: border-box;
	margin-top: 25px;
	width: 100%;
	padding: 18px;

	span {
		color: ${theme.engError};
	}
`;

const Title = styled.span`
	display: flex;
	font-size: 14px;
	justify-content: flex-start;
	align-items: center;

	span {
		padding-top: 4px;
	}

	img {
		margin-right: 10px;
	}
`;

const ListErrors = styled.ul`
	margin-top: 5px;
	margin-bottom: 5px;
	li {
		color: ${theme.engError};
	}
`;

function EngageFormErrors({ errors }) {
	if (!errors || _.isEmpty(errors)) {
		return <div />;
	}

	return (
		<Container>
			<Title>
				<img src={WarningIcon} alt="" />
				<span>
					<b>Whoops!</b> There were some problems with your input
				</span>
			</Title>
			<ListErrors>
				{_.map(errors, (value, key) => (
					<li key={key}>{value.value ? value.value : value} </li>
				))}
			</ListErrors>
		</Container>
	);
}

EngageFormErrors.propTypes = {
	errors: PropTypes.any,
};

export default React.memo(EngageFormErrors);
