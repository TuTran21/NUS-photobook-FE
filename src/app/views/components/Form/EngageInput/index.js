/* eslint-disable react/jsx-props-no-spreading */
import PropTypes from 'prop-types';
import React from 'react';
import classnames from 'classnames';
import styled from 'styled-components';
import theme from 'app/themes/styles';

const StyledFormGroup = styled.div`
	margin-bottom: 10px;
	&.errors.form-group {
		margin-bottom: 0px !important;
	}
	width: ${props => (props.width ? props.width : 'auto')};
`;

const Label = styled.label`
    display: inline-block;
    margin-bottom: 0 !important;
    cursor: pointer;
    line-height: 1.5;
    font-family: 'Nunito',Helvetica,Arial,sans-serif;
    font-size: 16px;
}`;

const StyledLabel = styled(Label)`
	font-size: 14px;
`;

const StyledError = styled(Label)`
	font-size: 14px;
	color: ${theme.ncpRed};
	float: right;
`;

const Input = styled.input`
	display: block;
	width: 100%;
	height: calc(1.5em + 0.75rem + 2px);
	padding: 0.375rem 0.75rem;
	font-size: 1rem;
	font-weight: 400;
	line-height: 1.5;
	color: #495057;
	background-color: ${theme.ncpWhite};
	background-clip: padding-box;
	border: 1px solid #ced4da;
	border-radius: 0.25rem;

	:focus {
		color: #495057;
		background-color: #fff;
		border-color: #80bdff;
		outline: 0;
		-webkit-box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
		box-shadow: 0 0 0 0.2rem rgba(0, 123, 255, 0.25);
	}
`;

const StyledInput = styled(Input)`
	${props => (props.errormessage ? `border: 1px solid ${theme.ncpRed};` : '')}
`;

function EngageInput({
	errors,
	title,
	required,
	value,
	maxLength,
	width,
	holderStyle,
	onChange,
	name,
	errorMessage,
	...restProps
}) {
	return (
		<StyledFormGroup
			width={width}
			className={classnames('eng-input--primary', {
				errors: !!errors,
			})}
			style={holderStyle}
		>
			{!!title && (
				<StyledLabel
					className={classnames('eng-label--input-title', {
						'eng-label--required': required,
					})}
				>
					{title}
				</StyledLabel>
			)}
			{errorMessage && <StyledError>{errorMessage}</StyledError>}
			<StyledInput
				name={name || 'email'}
				type="text"
				value={value || undefined}
				maxLength={maxLength || '100'}
				onChange={e => onChange(e)}
				errormessage={errorMessage}
				{...restProps}
			/>
		</StyledFormGroup>
	);
}

EngageInput.propTypes = {
	errors: PropTypes.any,
	title: PropTypes.any,
	value: PropTypes.any,
	required: PropTypes.any,
	maxLength: PropTypes.any,
};

export default EngageInput;
