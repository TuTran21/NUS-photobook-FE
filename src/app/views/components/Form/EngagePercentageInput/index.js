/* eslint-disable react/prop-types */
import React from 'react';
import NumberFormat from 'react-number-format';
import classnames from 'classnames';
import { FormGroup, Label } from 'reactstrap';

function EngagePercentageInput({ title, name, onChange, value, disabled, required }) {
	return (
		<FormGroup className="eng-input--primary">
			<Label
				className={classnames('eng-label--input-title', {
					'eng-label--required': required,
				})}
			>
				{title}
			</Label>
			<NumberFormat
				value={value}
				name={name}
				disabled={disabled}
				thousandSeparator=","
				placeholder="% here"
				className="form-control"
				suffix="%"
				onValueChange={event => {
					onChange(event.floatValue);
				}}
			/>
		</FormGroup>
	);
}

export default EngagePercentageInput;
