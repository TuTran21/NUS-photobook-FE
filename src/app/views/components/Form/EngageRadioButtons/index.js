/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-props-no-spreading */
import React, { useState, useCallback, useEffect } from 'react';
import styled from 'styled-components';
import theme from 'app/themes/styles';
import _ from 'lodash';
import classnames from 'classnames';
import UnChecked from './UnChecked.svg';
import Checked from './Checked.svg';

import { incrementAlphabet } from 'utils/utils';
import QuestionNumber from '../../Photo/QuestionNumber';
import UnCheckedSquare from './UnCheckedSquare.svg';
import CheckedSquare from './CheckedSquare.svg';
import LoadingIndicator from '../../common/LoadingIndicator';

const Wrapper = styled.div`
	display: flex;
	margin-bottom: 10px;

	${props => (props.flexvertical ? `flex-direction: column;` : '')}

	${props =>
		props.disabled
			? `
      background-color: ${theme.engDisabledBackground};
      border: none !important;
      cursor: not-allowed !important;
      opacity: 0.9;
  `
			: ''}
`;

const Label = styled.label`
    display: inline-block;
    margin-bottom: 0 !important;
    cursor: pointer;
    line-height: 1.5;
    font-size: 16px;
}`;

const StyledCheckbox = styled.div`
	display: flex;
	align-items: center;
	justify-content: flex-start;
	cursor: pointer;
	margin-right: 20px;
	height: 36px;
	margin-bottom: 8px;

	label {
		margin-bottom: 0 !important;
		margin-left: 5px;
		color: ${theme.ncpDark};
		cursor: pointer;
	}

	img {
		cursor: pointer;
		margin-top: -1px;
		width: 16px;
	}
`;

export default function({
	title,
	required,
	defaultValue,
	onChange,
	isMulti = false,
	disabled,
	flexvertical,
	holderStyle,
	hasAlphabetLabel,
	value,
	options = [],
}) {
	const [selected, setSelected] = useState(defaultValue || (isMulti ? [] : {}));

	const onSelect = useCallback(
		radioButton => {
			if (disabled) {
				return;
			}

			if (isMulti) {
				const isExistedIdx = _.findIndex(selected, sel => sel.key === radioButton.key);
				if (isExistedIdx > -1) {
					const newSelected = [...selected];
					newSelected.splice(isExistedIdx, 1);
					setSelected(newSelected);
					onChange(newSelected);
				} else {
					setSelected([...selected, radioButton]);
					onChange([...selected, radioButton]);
				}
			} else if (_.isEmpty(selected) || selected.key !== radioButton.key) {
				setSelected(radioButton);
				onChange(radioButton);
			} else {
				setSelected({});
				onChange(null);
			}
		},
		[selected, setSelected, onChange, disabled],
	);

	const isSelected = key => {
		if (_.isEmpty(selected)) {
			return false;
		}

		if (isMulti) {
			return !!_.find(selected, sel => sel.key === key);
		}

		return selected.key === key;
	};

	const isUnselected = key => {
		if (_.isEmpty(selected)) {
			return true;
		}

		if (isMulti) {
			return !_.find(selected, sel => sel.key === key);
		}

		return selected.key !== key;
	};

	return (
		<React.Fragment>
			{!!title && (
				<Label
					className={classnames('eng-label--input-title', {
						'eng-label--required': required,
					})}
				>
					{title}
				</Label>
			)}

			<Wrapper style={holderStyle} disabled={disabled} flexvertical={flexvertical}>
				{options.map((radioButton, idx) => (
					<StyledCheckbox key={idx} onClick={() => onSelect(radioButton)}>
						{hasAlphabetLabel && <QuestionNumber hasAlphabet>{incrementAlphabet(idx)}</QuestionNumber>}
						{isSelected(radioButton.key) && <img src={isMulti ? CheckedSquare : Checked} alt="" />}
						{isUnselected(radioButton.key) && <img src={isMulti ? UnCheckedSquare : UnChecked} alt="" />}
						<Label>{` ${radioButton.label}`}</Label>
					</StyledCheckbox>
				))}
			</Wrapper>
		</React.Fragment>
	);
}
