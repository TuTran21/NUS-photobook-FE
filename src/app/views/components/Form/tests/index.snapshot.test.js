import React from 'react';
import moment from 'moment';
import EngageCheckbox from 'components/Form/EngageCheckbox';
import EngageCurrencyInput from 'components/Form/EngageCurrencyInput';
import EngageDatePicker from 'components/Form/EngageDatePicker';
import EngageDropdown from 'components/Form/EngageDropdown';
// import EngageEditor from 'components/Form/EngageEditor';
import EngageFormErrors from 'components/Form/EngageFormErrors';
import EngageInput from 'components/Form/EngageInput';
import EngagePercentageInput from 'components/Form/EngagePercentageInput';
import EngageRadioButtons from 'components/Form/EngageRadioButtons';
import EngageSwitchYesNo from 'components/Form/EngageSwitchYesNo';
import EngageTagSelect from 'components/Form/EngageTagSelect';
import EngageTextarea from 'components/Form/EngageTextarea';

import renderer from 'react-test-renderer';

describe('components/Form', () => {
	it('EngageCheckbox should match snapshot', () => {
		const container = renderer
			.create(<EngageCheckbox disabled isChecked title="EngageCheckbox title" onChange={() => {}} />)
			.toJSON();

		expect(container).toMatchSnapshot();
	});

	it('EngageCurrencyInput should match snapshot', () => {
		const container = renderer
			.create(
				<EngageCurrencyInput
					required
					disabled
					currencyLabel="$"
					title="EngageCurrencyInput Title"
					value="100"
					name="EngageCurrencyInput_name"
					onChange={() => {}}
				/>,
			)
			.toJSON();

		expect(container).toMatchSnapshot();
	});

	it('EngageDatePicker should match snapshot', () => {
		const container = renderer
			.create(
				<EngageDatePicker
					required
					isClearable
					disabled
					title="EngageDatePicker__title"
					value={moment().toDate()}
					onChange={() => {}}
					minDate={moment().toDate()}
					maxDate={moment().toDate()}
				/>,
			)
			.toJSON();

		expect(container).toMatchSnapshot();
	});

	it('EngageDropdown should match snapshot', () => {
		const container = renderer
			.create(
				<EngageDropdown
					required
					disabled
					isClearable
					title="EngageDropdown__tite"
					value={undefined}
					onChange={() => {}}
					placeholder="EngageDropdown__placeholder"
					options={[]}
				/>,
			)
			.toJSON();

		expect(container).toMatchSnapshot();
	});

	it('EngageFormErrors should match snapshot', () => {
		const container = renderer
			.create(
				<EngageFormErrors
					errors={{
						error1: 'Error row 1',
						error2: 'Error row 2',
					}}
				/>,
			)
			.toJSON();

		expect(container).toMatchSnapshot();
	});

	it('EngageInput should match snapshot', () => {
		const container = renderer
			.create(
				<EngageInput
					required
					title="Last name"
					type="text"
					placeholder="Enter last name"
					value="EngageInput__value"
					onChange={() => {}}
				/>,
			)
			.toJSON();

		expect(container).toMatchSnapshot();
	});

	it('EngagePercentageInput should match snapshot', () => {
		const container = renderer
			.create(
				<EngagePercentageInput
					required
					disabled
					title="Sales fee"
					value="EngagePercentageInput__value"
					name="sales_charge"
					onChange={() => {}}
				/>,
			)
			.toJSON();

		expect(container).toMatchSnapshot();
	});

	it('EngageRadioButtons should match snapshot', () => {
		const container = renderer
			.create(
				<EngageRadioButtons
					required
					disabled
					title="Client restrictions"
					defaultValue={undefined}
					onChange={() => {}}
					options={[]}
				/>,
			)
			.toJSON();

		expect(container).toMatchSnapshot();
	});

	it('EngageSwitchYesNo should match snapshot', () => {
		const container = renderer.create(<EngageSwitchYesNo onToggle={() => {}} defaultValue />).toJSON();

		expect(container).toMatchSnapshot();
	});

	it('EngageTagSelect should match snapshot', () => {
		const container = renderer
			.create(
				<EngageTagSelect required disabled title="Tags" defaultValue={[]} options={[]} onChange={() => {}} />,
			)
			.toJSON();

		expect(container).toMatchSnapshot();
	});

	it('EngageTextarea should match snapshot', () => {
		const container = renderer
			.create(
				<EngageTextarea
					disabled
					title="External full description"
					placeholder=""
					rows={3}
					maxLength={300}
					name="description"
					value="EngageTextarea__description"
					onChange={() => {}}
				/>,
			)
			.toJSON();

		expect(container).toMatchSnapshot();
	});
});
