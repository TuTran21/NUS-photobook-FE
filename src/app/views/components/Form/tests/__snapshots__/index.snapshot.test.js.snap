// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Form EngageCheckbox should match snapshot 1`] = `
<div
  className="EngageCheckbox__StyledCheckbox-sc-1bflotg-0 dZKQKG form-group"
  onClick={[Function]}
>
  <img
    alt=""
    src="IMAGE_MOCK"
  />
  <label
    className=""
  >
     EngageCheckbox title
  </label>
</div>
`;

exports[`components/Form EngageCurrencyInput should match snapshot 1`] = `
<div
  className="eng-input--primary form-group"
>
  <label
    className="eng-label--input-title eng-label--required"
  >
    EngageCurrencyInput Title
  </label>
  <input
    className="form-control"
    disabled={true}
    name="EngageCurrencyInput_name"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    placeholder="$ here"
    type="text"
    value="$100"
  />
</div>
`;

exports[`components/Form EngageDatePicker should match snapshot 1`] = `
<div
  className="eng-input--primary form-group"
>
  <label
    className="eng-label--input-title eng-label--required"
  >
    EngageDatePicker__title
  </label>
  <div
    className="react-datepicker-wrapper"
  >
    <div
      className="react-datepicker__input-container"
    >
      <input
        className="form-control"
        disabled={true}
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Choose an option"
        readOnly={false}
        type="text"
        value="2020-01-19"
      />
    </div>
  </div>
  <img
    alt=""
    src="IMAGE_MOCK"
  />
</div>
`;

exports[`components/Form EngageDropdown should match snapshot 1`] = `
<div
  className="EngageDropdown__FormGroupStyled-sc-1kw1cd6-0 bDoKkd form-group"
>
  <label
    className="eng-label--input-title eng-label--required"
    htmlFor="EngageDropdown__tite"
  >
    EngageDropdown__tite
  </label>
  <div
    className="eng-select--primary EngageDropdown__StyledSelect-sc-1kw1cd6-1 jDqsr eng-select--primary--is-disabled css-14jk2my-container"
    onKeyDown={[Function]}
  >
    <div
      className="eng-select--primary__control eng-select--primary__control--is-disabled css-1fhf3k1-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className="eng-select--primary__value-container css-1hwfws3"
      >
        <div
          className="eng-select--primary__placeholder css-1c7vfki-Placeholder"
        >
          EngageDropdown__placeholder
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className="eng-select--primary__input"
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={true}
              id="react-select-2-input"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className="eng-select--primary__indicators css-1hb7zxy-IndicatorsContainer"
      >
        <span
          className="eng-select--primary__indicator-separator css-109onse-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          className="eng-select--primary__indicator eng-select--primary__dropdown-indicator css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <img
            alt=""
            src="IMAGE_MOCK"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`components/Form EngageFormErrors should match snapshot 1`] = `
<div
  className="EngageFormErrors__Container-tw85ct-0 UZjty"
>
  <span
    className="EngageFormErrors__Title-tw85ct-1 dpqUbb"
  >
    <img
      alt=""
      src="IMAGE_MOCK"
    />
    <span>
      <b>
        Whoops!
      </b>
       There were some problems with your input
    </span>
  </span>
  <ul
    className="EngageFormErrors__ListErrors-tw85ct-2 bziGkU"
  >
    <li>
      Error row 1
    </li>
    <li>
      Error row 2
    </li>
  </ul>
</div>
`;

exports[`components/Form EngageInput should match snapshot 1`] = `
<div
  className="eng-input--primary EngageInput__StyledFormGroup-sc-2i82me-0 kVVsnx form-group"
>
  <label
    className="eng-label--input-title eng-label--required"
  >
    Last name
  </label>
  <input
    className="form-control"
    maxLength="100"
    name="email"
    onChange={[Function]}
    placeholder="Enter last name"
    type="text"
    value="EngageInput__value"
  />
</div>
`;

exports[`components/Form EngagePercentageInput should match snapshot 1`] = `
<div
  className="eng-input--primary form-group"
>
  <label
    className="eng-label--input-title eng-label--required"
  >
    Sales fee
  </label>
  <input
    className="form-control"
    disabled={true}
    name="sales_charge"
    onBlur={[Function]}
    onChange={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseUp={[Function]}
    placeholder="% here"
    type="text"
    value=""
  />
</div>
`;

exports[`components/Form EngageRadioButtons should match snapshot 1`] = `
Array [
  <label
    className="eng-label--input-title eng-label--required"
  >
    Client restrictions
  </label>,
  <div
    className="EngageRadioButtons__Wrapper-lkavtw-0 iSXyax"
    disabled={true}
  />,
]
`;

exports[`components/Form EngageSwitchYesNo should match snapshot 1`] = `
<button
  className="EngageSwitchYesNo__StyledToggleButton-sc-185lsjz-0 fxVFHc"
  onClick={[Function]}
>
  
  <span>
    YES
  </span>
  <img
    alt=""
    src="IMAGE_MOCK"
  />
</button>
`;

exports[`components/Form EngageTagSelect should match snapshot 1`] = `
<div
  className="form-group"
>
  <label
    className="eng-label--input-title eng-label--required"
  >
    Tags
  </label>
  <div
    className="EngageTagSelect__SelectedTags-sc-1du8i3g-0 huFIiG"
  />
  <div
    className="EngageTagSelect__SearchBox-sc-1du8i3g-3 kEOzjl"
  >
    <img
      alt=""
      src="IMAGE_MOCK"
    />
    <input
      className="form-control"
      disabled={true}
      onChange={[Function]}
      placeholder="Type to search..."
      type="text"
    />
  </div>
  <div
    className="EngageTagSelect__ListTags-sc-1du8i3g-1 bKrMSv"
    disabled={true}
  />
</div>
`;

exports[`components/Form EngageTextarea should match snapshot 1`] = `
<div
  className="eng-input--primary EngageTextarea__StyledFormGroup-gbzf9b-0 iqOeQC form-group"
>
  <label
    className="eng-label--input-title EngageTextarea__FloatLabel-gbzf9b-1 SruCg"
  >
    273
     character
    <span>
      s
    </span>
     remaining
  </label>
  <label
    className="eng-label--input-title"
  >
    External full description
  </label>
  <textarea
    className="EngageTextarea__StyledInput-gbzf9b-2 jmZdrs form-control"
    disabled={true}
    maxLength={300}
    name="description"
    onChange={[Function]}
    placeholder=""
    rows={3}
    value="EngageTextarea__description"
  />
</div>
`;
